[0.10.4.1] : panic: runtime error: invalid memory address or nil pointer dereference

Hi,
I have telegraf deployed on around 1000servers, and It's crashing regulary on 1host (debian wheezy)


```
2016/03/09 09:12:38 Gathered metrics, (30s interval), from 12 inputs in 27.215759163s
panic: runtime error: invalid memory address or nil pointer dereference
	panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x6ed5c3]

goroutine 24574 [running]:
net.(*UnixConn).Close(0x0, 0x0, 0x0)
	<autogenerated>:118 +0x23
net.(*UnixConn).Write(0x0, 0xc82028735c, 0x10, 0x40, 0x0, 0x0, 0x0)
	<autogenerated>:117 +0x2c
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*conn).writeRecord(0xc820287320, 0xc800015a01, 0xc820935aa0, 0x8, 0x8, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm_fcgi.go:162 +0x418
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*conn).writeBeginRequest(0xc820287320, 0x7f0000010001, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm_fcgi.go:168 +0xbb
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*conn).Request(0xc820287320, 0xc820186a80, 0xc820935c38, 0xb, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm_fcgi.go:307 +0x130
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).gatherFcgi(0xc8200da180, 0xc820287320, 0xc820174e28, 0xa, 0x7f87d393ba40, 0xc820188000, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:152 +0x263
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).gatherServer(0xc8200da180, 0xc820174e11, 0x21, 0x7f87d393ba40, 0xc820188000, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:139 +0x6f5
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather.func1(0xc820376000, 0xc820376010, 0xc8200da180, 0x7f87d393ba40, 0xc820188000, 0xc820174e11, 0x21)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:89 +0x7b
created by github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:90 +0x1d1

goroutine 1 [semacquire]:
sync.runtime_Semacquire(0xc820aea04c)
	/usr/local/go/src/runtime/sema.go:43 +0x26
sync.(*WaitGroup).Wait(0xc820aea040)
	/usr/local/go/src/sync/waitgroup.go:126 +0xb4
github.com/influxdata/telegraf/agent.(*Agent).gatherParallel(0xc8201c4038, 0xc820154480, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:148 +0x145
github.com/influxdata/telegraf/agent.(*Agent).Run(0xc8201c4038, 0xc8201541e0, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:376 +0xa84
main.main()
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/cmd/telegraf/telegraf.go:234 +0x1cbc

goroutine 17 [syscall, 374 minutes, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1721 +0x1

goroutine 5 [syscall, 374 minutes]:
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x37

goroutine 70 [chan receive, 374 minutes]:
main.main.func2(0xc820154240, 0xc8201541e0, 0xc820456000)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/cmd/telegraf/telegraf.go:206 +0x47
created by main.main
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/cmd/telegraf/telegraf.go:216 +0x1589

goroutine 69 [select, 374 minutes, locked to thread]:
runtime.gopark(0x1253938, 0xc820033728, 0x104bd90, 0x6, 0xc820017018, 0x2)
	/usr/local/go/src/runtime/proc.go:185 +0x163
runtime.selectgoImpl(0xc820033728, 0x0, 0x18)
	/usr/local/go/src/runtime/select.go:392 +0xa64
runtime.selectgo(0xc820033728)
	/usr/local/go/src/runtime/select.go:212 +0x12
runtime.ensureSigM.func1()
	/usr/local/go/src/runtime/signal1_unix.go:227 +0x353
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1721 +0x1

goroutine 71 [runnable]:
github.com/influxdata/telegraf/agent.(*Agent).flusher(0xc8201c4038, 0xc8201541e0, 0xc820154480, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:275 +0x340
github.com/influxdata/telegraf/agent.(*Agent).Run.func1(0xc8201ae6c0, 0xc8201c4038, 0xc8201541e0, 0xc820154480)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:353 +0x7f
created by github.com/influxdata/telegraf/agent.(*Agent).Run
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:357 +0x938

goroutine 32 [IO wait]:
net.runtime_pollWait(0x7f87da01a198, 0x72, 0xc8200101d0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc82039a3e0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc82039a3e0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc82039a380, 0xc82031e000, 0x1000, 0x1000, 0x0, 0x7f87da00f050, 0xc8200101d0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc820afc010, 0xc82031e000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
net/http.noteEOFReader.Read(0x7f87da01a510, 0xc820afc010, 0xc820184f78, 0xc82031e000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:1370 +0x67
net/http.(*noteEOFReader).Read(0xc8200da0a0, 0xc82031e000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	<autogenerated>:126 +0xd0
bufio.(*Reader).fill(0xc820018180)
	/usr/local/go/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Peek(0xc820018180, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:132 +0xcc
net/http.(*persistConn).readLoop(0xc820184f20)
	/usr/local/go/src/net/http/transport.go:876 +0xf7
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:685 +0xc78

goroutine 33636 [semacquire]:
sync.runtime_Semacquire(0xc820aca00c)
	/usr/local/go/src/runtime/sema.go:43 +0x26
sync.(*WaitGroup).Wait(0xc820aca000)
	/usr/local/go/src/sync/waitgroup.go:126 +0xb4
github.com/influxdata/telegraf/plugins/inputs/apache.(*Apache).Gather(0xc8200da100, 0x7f87d393ba40, 0xc820bfa000, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/apache/apache.go:52 +0x379
github.com/influxdata/telegraf/agent.(*Agent).gatherParallel.func1(0xc820aea040, 0xc820154480, 0xc8201c4038, 0x0, 0xc820186840)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:137 +0x432
created by github.com/influxdata/telegraf/agent.(*Agent).gatherParallel
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:141 +0x442

goroutine 24572 [runnable]:
time.Sleep(0x186a0)
	/usr/local/go/src/runtime/time.go:59 +0xf9
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*conn).writePairs(0xc820286e10, 0x7f0000010004, 0xc820186a20, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm_fcgi.go:179 +0x6df
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*conn).Request(0xc820286e10, 0xc820186a20, 0xc8208fdc38, 0xb, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm_fcgi.go:312 +0x194
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).gatherFcgi(0xc8200da180, 0xc820286e10, 0xc820174dc8, 0xa, 0x7f87d393ba40, 0xc820188000, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:152 +0x263
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).gatherServer(0xc8200da180, 0xc820174db1, 0x21, 0x7f87d393ba40, 0xc820188000, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:139 +0x6f5
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather.func1(0xc820376000, 0xc820376010, 0xc8200da180, 0x7f87d393ba40, 0xc820188000, 0xc820174db1, 0x21)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:89 +0x7b
created by github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:90 +0x1d1

goroutine 33 [select]:
net/http.(*persistConn).writeLoop(0xc820184f20)
	/usr/local/go/src/net/http/transport.go:1009 +0x40c
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:686 +0xc9d

goroutine 24573 [runnable]:
net.runtime_pollWait(0x7f87da019b98, 0x72, 0xc8200101d0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc820bf0060, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820bf0060, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc820bf0000, 0xc820345220, 0x8, 0x8, 0x0, 0x7f87da00f050, 0xc8200101d0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc820396020, 0xc820345220, 0x8, 0x8, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
io.ReadAtLeast(0x7f87d307a1e0, 0xc820396020, 0xc820345220, 0x8, 0x8, 0x8, 0x0, 0x0, 0x0)
	/usr/local/go/src/io/io.go:298 +0xe6
io.ReadFull(0x7f87d307a1e0, 0xc820396020, 0xc820345220, 0x8, 0x8, 0x8, 0x0, 0x0)
	/usr/local/go/src/io/io.go:316 +0x62
encoding/binary.Read(0x7f87d307a1e0, 0xc820396020, 0x7f87d307a138, 0x1795760, 0xe82760, 0xc820688000, 0x0, 0x0)
	/usr/local/go/src/encoding/binary/binary.go:216 +0x1336
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*record).read(0xc820688000, 0x7f87d307a1e0, 0xc820396020, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm_fcgi.go:130 +0xdd
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*conn).Request(0xc8202862d0, 0xc820186a50, 0xc820389c38, 0xb, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm_fcgi.go:329 +0x32b
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).gatherFcgi(0xc8200da180, 0xc8202862d0, 0xc820174df8, 0xa, 0x7f87d393ba40, 0xc820188000, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:152 +0x263
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).gatherServer(0xc8200da180, 0xc820174de1, 0x21, 0x7f87d393ba40, 0xc820188000, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:139 +0x6f5
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather.func1(0xc820376000, 0xc820376010, 0xc8200da180, 0x7f87d393ba40, 0xc820188000, 0xc820174de1, 0x21)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:89 +0x7b
created by github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:90 +0x1d1

goroutine 24577 [runnable]:
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*conn).Request(0xc820973b00, 0xc82026fbf0, 0xc8205bdc38, 0xb, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm_fcgi.go:329 +0x305
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).gatherFcgi(0xc8200da180, 0xc820973b00, 0xc820174eb8, 0xa, 0x7f87d393ba40, 0xc820188000, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:152 +0x263
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).gatherServer(0xc8200da180, 0xc820174ea1, 0x21, 0x7f87d393ba40, 0xc820188000, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:139 +0x6f5
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather.func1(0xc820376000, 0xc820376010, 0xc8200da180, 0x7f87d393ba40, 0xc820188000, 0xc820174ea1, 0x21)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:89 +0x7b
created by github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:90 +0x1d1

goroutine 33690 [runnable]:
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather.func1(0xc820376000, 0xc820376010, 0xc8200da180, 0x7f87d393ba40, 0xc820188000, 0xc820186031, 0x21)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:87
created by github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:90 +0x1d1

goroutine 33641 [runnable]:
syscall.Syscall(0x3, 0x18, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/syscall/asm_linux_amd64.s:18 +0x5
syscall.Close(0x18, 0x0, 0x0)
	/usr/local/go/src/syscall/zsyscall_linux_amd64.go:296 +0x4a
syscall.forkExec(0x1053580, 0xd, 0xc820b9a4c0, 0x4, 0x4, 0xc8205c1398, 0x2970, 0x0, 0x0)
	/usr/local/go/src/syscall/exec_unix.go:201 +0x540
syscall.StartProcess(0x1053580, 0xd, 0xc820b9a4c0, 0x4, 0x4, 0xc8205c1398, 0x2, 0x4, 0x0, 0x0)
	/usr/local/go/src/syscall/exec_unix.go:240 +0x6b
os.startProcess(0x1053580, 0xd, 0xc820b9a4c0, 0x4, 0x4, 0xc8205c15d0, 0xc820b52000, 0x0, 0x0)
	/usr/local/go/src/os/exec_posix.go:45 +0x33c
os.StartProcess(0x1053580, 0xd, 0xc820b9a4c0, 0x4, 0x4, 0xc8205c15d0, 0x0, 0x0, 0x0)
	/usr/local/go/src/os/doc.go:24 +0x69
os/exec.(*Cmd).Start(0xc82017a280, 0x0, 0x0)
	/usr/local/go/src/os/exec/exec.go:328 +0x767
os/exec.(*Cmd).Run(0xc82017a280, 0x0, 0x0)
	/usr/local/go/src/os/exec/exec.go:255 +0x2d
os/exec.(*Cmd).Output(0xc82017a280, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/os/exec/exec.go:408 +0x1a1
github.com/shirou/gopsutil/disk.GetDiskSerialNumber(0xc82039a2ad, 0x4, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/shirou/gopsutil/disk/disk_linux.go:302 +0x212
github.com/shirou/gopsutil/disk.DiskIOCounters(0x0, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/shirou/gopsutil/disk/disk_linux.go:294 +0x774
github.com/influxdata/telegraf/plugins/inputs/system.(*systemPS).DiskIO(0x1795760, 0x7f87da1c0c50, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/system/ps.go:116 +0x28
github.com/influxdata/telegraf/plugins/inputs/system.(*DiskIOStats).Gather(0xc820187710, 0x7f87d393ba40, 0xc820360c40, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/system/disk.go:104 +0x67
github.com/influxdata/telegraf/agent.(*Agent).gatherParallel.func1(0xc820aea040, 0xc820154480, 0xc8201c4038, 0x0, 0xc820187770)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:137 +0x432
created by github.com/influxdata/telegraf/agent.(*Agent).gatherParallel
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:141 +0x442

goroutine 33637 [semacquire]:
sync.runtime_Semacquire(0xc82037600c)
	/usr/local/go/src/runtime/sema.go:43 +0x26
sync.(*WaitGroup).Wait(0xc820376000)
	/usr/local/go/src/sync/waitgroup.go:126 +0xb4
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather(0xc8200da180, 0x7f87d393ba40, 0xc820188000, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:93 +0x1fe
github.com/influxdata/telegraf/agent.(*Agent).gatherParallel.func1(0xc820aea040, 0xc820154480, 0xc8201c4038, 0x0, 0xc820186cc0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:137 +0x432
created by github.com/influxdata/telegraf/agent.(*Agent).gatherParallel
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:141 +0x442

goroutine 33686 [runnable]:
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather.func1(0xc820376000, 0xc820376010, 0xc8200da180, 0x7f87d393ba40, 0xc820188000, 0xc820174f91, 0x21)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:87
created by github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:90 +0x1d1

goroutine 33694 [runnable]:
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather.func1(0xc820376000, 0xc820376010, 0xc8200da180, 0x7f87d393ba40, 0xc820188000, 0xc820186571, 0x21)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:87
created by github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:90 +0x1d1

goroutine 33692 [runnable]:
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather.func1(0xc820376000, 0xc820376010, 0xc8200da180, 0x7f87d393ba40, 0xc820188000, 0xc820186091, 0x21)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:87
created by github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:90 +0x1d1

goroutine 33544 [syscall]:
syscall.Syscall(0x0, 0x11, 0xc820b60e00, 0x200, 0x40040, 0x2e220, 0x412749)
	/usr/local/go/src/syscall/asm_linux_amd64.s:18 +0x5
syscall.read(0x11, 0xc820b60e00, 0x200, 0x200, 0x0, 0x0, 0x0)
	/usr/local/go/src/syscall/zsyscall_linux_amd64.go:783 +0x5f
syscall.Read(0x11, 0xc820b60e00, 0x200, 0x200, 0xc820989ce8, 0x0, 0x0)
	/usr/local/go/src/syscall/syscall_unix.go:160 +0x4d
os.(*File).read(0xc82003a0f8, 0xc820b60e00, 0x200, 0x200, 0xc820b60e00, 0x0, 0x0)
	/usr/local/go/src/os/file_unix.go:211 +0x53
os.(*File).Read(0xc82003a0f8, 0xc820b60e00, 0x200, 0x200, 0x58b086, 0x0, 0x0)
	/usr/local/go/src/os/file.go:95 +0x8a
bytes.(*Buffer).ReadFrom(0xc8200d0150, 0x7f87da013298, 0xc82003a0f8, 0x0, 0x0, 0x0)
	/usr/local/go/src/bytes/buffer.go:173 +0x23f
io.copyBuffer(0x7f87da014120, 0xc8200d0150, 0x7f87da013298, 0xc82003a0f8, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/io/io.go:375 +0x180
io.Copy(0x7f87da014120, 0xc8200d0150, 0x7f87da013298, 0xc82003a0f8, 0x0, 0x0, 0x0)
	/usr/local/go/src/io/io.go:351 +0x64
os/exec.(*Cmd).writerDescriptor.func1(0x0, 0x0)
	/usr/local/go/src/os/exec/exec.go:232 +0x8b
os/exec.(*Cmd).Start.func1(0xc820aee280, 0xc8205674a0)
	/usr/local/go/src/os/exec/exec.go:340 +0x1d
created by os/exec.(*Cmd).Start
	/usr/local/go/src/os/exec/exec.go:341 +0x96d

goroutine 24575 [runnable]:
reflect.Value.Field(0xf7b6c0, 0xc820345570, 0x59, 0x5, 0x0, 0x0, 0x7f87da0130a8)
	/usr/local/go/src/reflect/value.go:737
encoding/binary.(*encoder).value(0xc8205bf9a0, 0xf7b6c0, 0xc820345570, 0x59)
	/usr/local/go/src/encoding/binary/binary.go:517 +0x9e5
encoding/binary.Write(0x7f87da014120, 0xc820287968, 0x7f87d307a138, 0x1795760, 0xf7b6c0, 0xc820345570, 0x0, 0x0)
	/usr/local/go/src/encoding/binary/binary.go:316 +0x1792
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*conn).writeRecord(0xc820287950, 0xc80001fc05, 0xc8205bfb10, 0xb, 0x20, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm_fcgi.go:153 +0x21b
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*conn).Request(0xc820287950, 0xc820186ab0, 0xc8205bfc38, 0xb, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm_fcgi.go:318 +0x253
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).gatherFcgi(0xc8200da180, 0xc820287950, 0xc820174e58, 0xa, 0x7f87d393ba40, 0xc820188000, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:152 +0x263
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).gatherServer(0xc8200da180, 0xc820174e41, 0x21, 0x7f87d393ba40, 0xc820188000, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:139 +0x6f5
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather.func1(0xc820376000, 0xc820376010, 0xc8200da180, 0x7f87d393ba40, 0xc820188000, 0xc820174e41, 0x21)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:89 +0x7b
created by github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:90 +0x1d1

goroutine 33691 [runnable]:
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather.func1(0xc820376000, 0xc820376010, 0xc8200da180, 0x7f87d393ba40, 0xc820188000, 0xc820186061, 0x21)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:87
created by github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:90 +0x1d1

goroutine 33695 [runnable]:
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather.func1(0xc820376000, 0xc820376010, 0xc8200da180, 0x7f87d393ba40, 0xc820188000, 0xc8201866f1, 0x21)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:87
created by github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:90 +0x1d1

goroutine 33693 [runnable]:
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather.func1(0xc820376000, 0xc820376010, 0xc8200da180, 0x7f87d393ba40, 0xc820188000, 0xc8201860c1, 0x21)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:87
created by github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:90 +0x1d1

goroutine 33696 [runnable]:
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*conn).writePairs(0xc820c49320, 0x7f0000010004, 0xc820a7c5a0, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm_fcgi.go:179 +0x716
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*conn).Request(0xc820c49320, 0xc820a7c5a0, 0xc820a93c38, 0xb, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm_fcgi.go:312 +0x194
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).gatherFcgi(0xc8200da180, 0xc820c49320, 0xc820186738, 0xa, 0x7f87d393ba40, 0xc820188000, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:152 +0x263
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).gatherServer(0xc8200da180, 0xc820186721, 0x21, 0x7f87d393ba40, 0xc820188000, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:139 +0x6f5
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather.func1(0xc820376000, 0xc820376010, 0xc8200da180, 0x7f87d393ba40, 0xc820188000, 0xc820186721, 0x21)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:89 +0x7b
created by github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:90 +0x1d1

goroutine 33589 [select]:
net/http.(*Transport).getConn(0x1751640, 0xc8200a62a0, 0x0, 0xc82093c450, 0x4, 0xc8201b8000, 0x16, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:554 +0x663
net/http.(*Transport).RoundTrip(0x1751640, 0xc8200a62a0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:228 +0x4ac
net/http.send(0xc8200a62a0, 0x7f87da0134d8, 0x1751640, 0xc82093ca50, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:220 +0x52c
net/http.(*Client).send(0x176dee0, 0xc8200a62a0, 0x2d, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:143 +0x15a
net/http.(*Client).doFollowingRedirects(0x176dee0, 0xc8200a62a0, 0x12534f0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:380 +0xbc0
net/http.(*Client).Get(0x176dee0, 0xc82093c450, 0x2d, 0xffffffffffffffff, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:306 +0xaa
github.com/influxdata/telegraf/plugins/inputs/apache.(*Apache).gatherUrl(0xc8200da100, 0xc820288000, 0x7f87d393ba40, 0xc820bfa000, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/apache/apache.go:64 +0x79
github.com/influxdata/telegraf/plugins/inputs/apache.(*Apache).Gather.func1(0xc820aca000, 0xc820aca010, 0xc8200da100, 0x7f87d393ba40, 0xc820bfa000, 0xc820288000)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/apache/apache.go:48 +0x6a
created by github.com/influxdata/telegraf/plugins/inputs/apache.(*Apache).Gather
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/apache/apache.go:49 +0x34c

goroutine 33644 [runnable]:
github.com/influxdata/telegraf/plugins/inputs/system.(*NetIOStats).Gather(0xc82012a1e0, 0x7f87d393ba40, 0xc8209323c0, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/system/net.go:95 +0x793
github.com/influxdata/telegraf/agent.(*Agent).gatherParallel.func1(0xc820aea040, 0xc820154480, 0xc8201c4038, 0x0, 0xc82012a240)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:137 +0x432
created by github.com/influxdata/telegraf/agent.(*Agent).gatherParallel
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:141 +0x442

goroutine 33687 [runnable]:
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather.func1(0xc820376000, 0xc820376010, 0xc8200da180, 0x7f87d393ba40, 0xc820188000, 0xc820174fc1, 0x21)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:87
created by github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:90 +0x1d1

goroutine 33645 [syscall]:
syscall.Syscall6(0x3d, 0x296f, 0xc820b753bc, 0x0, 0xc820c49200, 0x0, 0x0, 0xc820b75390, 0xc820b75398, 0xc820b75388)
	/usr/local/go/src/syscall/asm_linux_amd64.s:44 +0x5
syscall.wait4(0x296f, 0xc820b753bc, 0x0, 0xc820c49200, 0x90, 0x0, 0x0)
	/usr/local/go/src/syscall/zsyscall_linux_amd64.go:172 +0x72
syscall.Wait4(0x296f, 0xc820b75404, 0x0, 0xc820c49200, 0xc82003a148, 0x0, 0x0)
	/usr/local/go/src/syscall/syscall_linux.go:256 +0x55
os.(*Process).wait(0xc820aa0460, 0x15, 0x0, 0x0)
	/usr/local/go/src/os/exec_unix.go:22 +0x105
os.(*Process).Wait(0xc820aa0460, 0x0, 0x0, 0x0)
	/usr/local/go/src/os/doc.go:45 +0x2d
os/exec.(*Cmd).Wait(0xc820aee280, 0x0, 0x0)
	/usr/local/go/src/os/exec/exec.go:380 +0x211
os/exec.(*Cmd).Run(0xc820aee280, 0x0, 0x0)
	/usr/local/go/src/os/exec/exec.go:258 +0x64
os/exec.(*Cmd).Output(0xc820aee280, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/os/exec/exec.go:408 +0x1a1
github.com/shirou/gopsutil/internal/common.Invoke.Command(0xc820aea240, 0xd, 0xc8200d00e0, 0x7, 0x7, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/shirou/gopsutil/internal/common/common.go:31 +0x7b
github.com/shirou/gopsutil/internal/common.(*Invoke).Command(0x1795760, 0xc820aea240, 0xd, 0xc8200d00e0, 0x7, 0x7, 0x0, 0x0, 0x0, 0x0, ...)
	<autogenerated>:1 +0xee
github.com/shirou/gopsutil/internal/common.CallLsof(0x7f87da014170, 0x1795760, 0x0, 0xc82018ac40, 0x4, 0x4, 0x0, 0x0, 0x0, 0x0, ...)
	/home/ubuntu/telegraf-build/src/github.com/shirou/gopsutil/internal/common/common_unix.go:23 +0x342
github.com/shirou/gopsutil/net.NetConnectionsPid(0x1045f70, 0x3, 0xc800000000, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/shirou/gopsutil/net/net_unix.go:50 +0x2ab
github.com/shirou/gopsutil/net.NetConnections(0x1045f70, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/shirou/gopsutil/net/net_unix.go:13 +0x50
github.com/influxdata/telegraf/plugins/inputs/system.(*systemPS).NetConnections(0x1795760, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/system/ps.go:112 +0x49
github.com/influxdata/telegraf/plugins/inputs/system.(*NetStats).Gather(0xc8201ae140, 0x7f87d393ba40, 0xc82018ac00, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/system/netstat.go:26 +0x76
github.com/influxdata/telegraf/agent.(*Agent).gatherParallel.func1(0xc820aea040, 0xc820154480, 0xc8201c4038, 0x0, 0xc82012a540)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:137 +0x432
created by github.com/influxdata/telegraf/agent.(*Agent).gatherParallel
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:141 +0x442

goroutine 33682 [runnable]:
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather.func1(0xc820376000, 0xc820376010, 0xc8200da180, 0x7f87d393ba40, 0xc820188000, 0xc820174ed1, 0x21)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:87
created by github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:90 +0x1d1

goroutine 33688 [runnable]:
net.runtime_pollWait(0x7f87da019d18, 0x72, 0xc8200101d0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc820205a30, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820205a30, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8202059d0, 0xc820bf3580, 0x8, 0x8, 0x0, 0x7f87da00f050, 0xc8200101d0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc8208b80f8, 0xc820bf3580, 0x8, 0x8, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
io.ReadAtLeast(0x7f87d307a1e0, 0xc8208b80f8, 0xc820bf3580, 0x8, 0x8, 0x8, 0x0, 0x0, 0x0)
	/usr/local/go/src/io/io.go:298 +0xe6
io.ReadFull(0x7f87d307a1e0, 0xc8208b80f8, 0xc820bf3580, 0x8, 0x8, 0x8, 0x0, 0x0)
	/usr/local/go/src/io/io.go:316 +0x62
encoding/binary.Read(0x7f87d307a1e0, 0xc8208b80f8, 0x7f87d307a138, 0x1795760, 0xe82760, 0xc820820000, 0x0, 0x0)
	/usr/local/go/src/encoding/binary/binary.go:216 +0x1336
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*record).read(0xc820820000, 0x7f87d307a1e0, 0xc8208b80f8, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm_fcgi.go:130 +0xdd
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*conn).Request(0xc820973dd0, 0xc82026fbc0, 0xc820903c38, 0xb, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm_fcgi.go:329 +0x32b
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).gatherFcgi(0xc8200da180, 0xc820973dd0, 0xc820175008, 0xa, 0x7f87d393ba40, 0xc820188000, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:152 +0x263
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).gatherServer(0xc8200da180, 0xc820174ff1, 0x21, 0x7f87d393ba40, 0xc820188000, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:139 +0x6f5
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather.func1(0xc820376000, 0xc820376010, 0xc8200da180, 0x7f87d393ba40, 0xc820188000, 0xc820174ff1, 0x21)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:89 +0x7b
created by github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:90 +0x1d1

goroutine 33684 [runnable]:
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather.func1(0xc820376000, 0xc820376010, 0xc8200da180, 0x7f87d393ba40, 0xc820188000, 0xc820174f31, 0x21)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:87
created by github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:90 +0x1d1

goroutine 33643 [runnable]:
net.runtime_pollWait(0x7f87da019598, 0x77, 0xc820a1a3c0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc82052e760, 0x77, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitWrite(0xc82052e760, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:82 +0x36
net.(*netFD).connect(0xc82052e700, 0x0, 0x0, 0x7f87da019330, 0xc820ae6800, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/fd_unix.go:114 +0x1f6
net.(*netFD).dial(0xc82052e700, 0x7f87da01a4c8, 0x0, 0x7f87da01a4c8, 0xc820a50390, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/sock_posix.go:137 +0x351
net.socket(0xc8200e1455, 0x3, 0x2, 0x1, 0x0, 0xc820a50300, 0x7f87da01a4c8, 0x0, 0x7f87da01a4c8, 0xc820a50390, ...)
	/usr/local/go/src/net/sock_posix.go:89 +0x411
net.internetSocket(0xc8200e1455, 0x3, 0x7f87da01a4c8, 0x0, 0x7f87da01a4c8, 0xc820a50390, 0x0, 0xc800000000, 0x0, 0x1, ...)
	/usr/local/go/src/net/ipsock_posix.go:160 +0x141
net.dialTCP(0xc8200e1455, 0x3, 0x0, 0xc820a50390, 0x0, 0xc800000000, 0x0, 0xc820a1a3b0, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:171 +0x11e
net.dialSingle(0xc820c4a100, 0x7f87da01a438, 0xc820a50390, 0x0, 0x7fff00000000, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/dial.go:364 +0x3f5
net.dialSerial.func1(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/dial.go:336 +0x75
net.dial(0xc8200e1455, 0x3, 0x7f87da01a438, 0xc820a50390, 0xc82021b4f0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/usr/local/go/src/net/fd_unix.go:40 +0x60
net.dialSerial(0xc820c4a100, 0xc820a1a3b0, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/dial.go:338 +0x760
net.(*Dialer).Dial(0xc82021b868, 0xc8200e1455, 0x3, 0xc8200e1459, 0xe, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/dial.go:232 +0x50f
github.com/go-sql-driver/mysql.MySQLDriver.Open(0xc8200e1441, 0x28, 0x0, 0x0, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/go-sql-driver/mysql/driver.go:68 +0x99c
github.com/go-sql-driver/mysql.(*MySQLDriver).Open(0x1795760, 0xc8200e1441, 0x28, 0x0, 0x0, 0x0, 0x0)
	<autogenerated>:6 +0xb9
database/sql.(*DB).conn(0xc8206186e0, 0xc820272a01, 0xc820272a90, 0x0, 0x0)
	/usr/local/go/src/database/sql/sql.go:710 +0x451
database/sql.(*DB).query(0xc8206186e0, 0x1132d60, 0x1e, 0x0, 0x0, 0x0, 0xc820272b01, 0xc820272b70, 0x0, 0x0)
	/usr/local/go/src/database/sql/sql.go:959 +0x46
database/sql.(*DB).Query(0xc8206186e0, 0x1132d60, 0x1e, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/database/sql/sql.go:947 +0xa3
github.com/influxdata/telegraf/plugins/inputs/mysql.(*Mysql).gatherServer(0xc82000fa60, 0xc8200e1441, 0x28, 0x7f87d393ba40, 0xc820603700, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/mysql/mysql.go:132 +0x20e
github.com/influxdata/telegraf/plugins/inputs/mysql.(*Mysql).Gather(0xc82000fa60, 0x7f87d393ba40, 0xc820603700, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/mysql/mysql.go:48 +0x12f
github.com/influxdata/telegraf/agent.(*Agent).gatherParallel.func1(0xc820aea040, 0xc820154480, 0xc8201c4038, 0x0, 0xc8200e1620)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:137 +0x432
created by github.com/influxdata/telegraf/agent.(*Agent).gatherParallel
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:141 +0x442

goroutine 33689 [runnable]:
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather.func1(0xc820376000, 0xc820376010, 0xc8200da180, 0x7f87d393ba40, 0xc820188000, 0xc820186001, 0x21)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:87
created by github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:90 +0x1d1

goroutine 33683 [runnable]:
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather.func1(0xc820376000, 0xc820376010, 0xc8200da180, 0x7f87d393ba40, 0xc820188000, 0xc820174f01, 0x21)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:87
created by github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:90 +0x1d1

goroutine 33638 [runnable]:
syscall.Syscall6(0x37, 0xe, 0x1, 0x4, 0xc820218f4c, 0xc820218f48, 0x0, 0x0, 0x4, 0x0)
	/usr/local/go/src/syscall/asm_linux_amd64.s:44 +0x5
syscall.getsockopt(0xe, 0x1, 0x4, 0xc820218f4c, 0xc820218f48, 0x0, 0x0)
	/usr/local/go/src/syscall/zsyscall_linux_amd64.go:1645 +0x73
syscall.GetsockoptInt(0xe, 0x1, 0x4, 0x0, 0x0, 0x0)
	/usr/local/go/src/syscall/syscall_unix.go:237 +0x67
net.(*netFD).connect(0xc82039a620, 0x0, 0x0, 0x7f87da019330, 0xc820625600, 0xece71d1fe, 0xf6df243, 0x176ea00, 0x0, 0x0)
	/usr/local/go/src/net/fd_unix.go:117 +0x262
net.(*netFD).dial(0xc82039a620, 0x7f87da01a4c8, 0x0, 0x7f87da01a4c8, 0xc820a7c030, 0xece71d1fe, 0xf6df243, 0x176ea00, 0x0, 0x0)
	/usr/local/go/src/net/sock_posix.go:137 +0x351
net.socket(0x104c950, 0x3, 0x2, 0x1, 0x0, 0xc820a7c000, 0x7f87da01a4c8, 0x0, 0x7f87da01a4c8, 0xc820a7c030, ...)
	/usr/local/go/src/net/sock_posix.go:89 +0x411
net.internetSocket(0x104c950, 0x3, 0x7f87da01a4c8, 0x0, 0x7f87da01a4c8, 0xc820a7c030, 0xece71d1fe, 0xc80f6df243, 0x176ea00, 0x1, ...)
	/usr/local/go/src/net/ipsock_posix.go:160 +0x141
net.dialTCP(0x104c950, 0x3, 0x0, 0xc820a7c030, 0xece71d1fe, 0xc80f6df243, 0x176ea00, 0x1, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:171 +0x11e
net.dialSingle(0xc820318800, 0x7f87da01a438, 0xc820a7c030, 0xece71d1fe, 0x7fff0f6df243, 0x176ea00, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/dial.go:364 +0x3f5
net.dialSerial.func1(0xece71d1fe, 0xf6df243, 0x176ea00, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/dial.go:336 +0x75
net.dial(0x104c950, 0x3, 0x7f87da01a438, 0xc820a7c030, 0xc820219790, 0xece71d1fe, 0xf6df243, 0x176ea00, 0x0, 0x0, ...)
	/usr/local/go/src/net/fd_unix.go:40 +0x60
net.dialSerial(0xc820318800, 0xc820c45330, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/dial.go:338 +0x760
net.(*Dialer).Dial(0xc820219a08, 0x104c950, 0x3, 0xc8200da0e1, 0xf, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/dial.go:232 +0x50f
net.DialTimeout(0x104c950, 0x3, 0xc8200da0e1, 0xf, 0x12a05f200, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/dial.go:193 +0xa2
github.com/influxdata/telegraf/plugins/inputs/memcached.(*Memcached).gatherServer(0xc820186cf0, 0xc8200da0e1, 0xf, 0x0, 0x7f87d393ba40, 0xc820a58b80, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/memcached/memcached.go:109 +0xdb1
github.com/influxdata/telegraf/plugins/inputs/memcached.(*Memcached).Gather(0xc820186cf0, 0x7f87d393ba40, 0xc820a58b80, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/memcached/memcached.go:77 +0x158
github.com/influxdata/telegraf/agent.(*Agent).gatherParallel.func1(0xc820aea040, 0xc820154480, 0xc8201c4038, 0x0, 0xc820186d50)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:137 +0x432
created by github.com/influxdata/telegraf/agent.(*Agent).gatherParallel
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:141 +0x442

goroutine 33685 [runnable]:
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather.func1(0xc820376000, 0xc820376010, 0xc8200da180, 0x7f87d393ba40, 0xc820188000, 0xc820174f61, 0x21)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:87
created by github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:90 +0x1d1

goroutine 33640 [runnable]:
github.com/influxdata/telegraf/agent.(*accumulator).AddFields(0xc8202225c0, 0x10473e0, 0x4, 0xc820a50b40, 0xc820a50b10, 0x0, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/accumulator.go:115 +0x1255
github.com/influxdata/telegraf/plugins/inputs/system.(*DiskStats).Gather(0xc820394060, 0x7f87d393ba40, 0xc8202225c0, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/system/disk.go:73 +0x974
github.com/influxdata/telegraf/agent.(*Agent).gatherParallel.func1(0xc820aea040, 0xc820154480, 0xc8201c4038, 0x0, 0xc820187440)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:137 +0x432
created by github.com/influxdata/telegraf/agent.(*Agent).gatherParallel
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:141 +0x442

goroutine 24571 [runnable]:
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*conn).writePairs(0xc820286a20, 0x7f0000010004, 0xc8201869f0, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm_fcgi.go:179 +0x716
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*conn).Request(0xc820286a20, 0xc8201869f0, 0xc820937c38, 0xb, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm_fcgi.go:312 +0x194
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).gatherFcgi(0xc8200da180, 0xc820286a20, 0xc820174d97, 0xa, 0x7f87d393ba40, 0xc820188000, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:152 +0x263
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).gatherServer(0xc8200da180, 0xc820174d81, 0x20, 0x7f87d393ba40, 0xc820188000, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:139 +0x6f5
github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather.func1(0xc820376000, 0xc820376010, 0xc8200da180, 0x7f87d393ba40, 0xc820188000, 0xc820174d81, 0x20)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:89 +0x7b
created by github.com/influxdata/telegraf/plugins/inputs/phpfpm.(*phpfpm).Gather
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/phpfpm/phpfpm.go:90 +0x1d1

goroutine 33639 [runnable]:
github.com/shirou/gopsutil/internal/common.ReadLinesOffsetN(0xc820b1a010, 0xa, 0xe, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/shirou/gopsutil/internal/common/common.go:89 +0x460
github.com/shirou/gopsutil/cpu.CPUTimes(0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/shirou/gopsutil/cpu/cpu_linux.go:34 +0x135
github.com/influxdata/telegraf/plugins/inputs/system.(*systemPS).CPUTimes(0x1795760, 0x7f87d3070101, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/system/ps.go:38 +0x82
github.com/influxdata/telegraf/plugins/inputs/system.(*CPUStats).Gather(0xc8201870e0, 0x7f87d393ba40, 0xc82042e0c0, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/system/cpu.go:44 +0x88
github.com/influxdata/telegraf/agent.(*Agent).gatherParallel.func1(0xc820aea040, 0xc820154480, 0xc8201c4038, 0x0, 0xc820187140)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:137 +0x432
created by github.com/influxdata/telegraf/agent.(*Agent).gatherParallel
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:141 +0x442

goroutine 33653 [chan receive]:
database/sql.(*DB).connectionOpener(0xc8206186e0)
	/usr/local/go/src/database/sql/sql.go:634 +0x45
created by database/sql.Open
	/usr/local/go/src/database/sql/sql.go:481 +0x336

goroutine 33652 [IO wait]:
net.runtime_pollWait(0x7f87da0194d8, 0x77, 0xc820a1a390)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc82052e530, 0x77, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitWrite(0xc82052e530, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:82 +0x36
net.(*netFD).connect(0xc82052e4d0, 0x0, 0x0, 0x7f87da019330, 0xc820ae67e0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/fd_unix.go:114 +0x1f6
net.(*netFD).dial(0xc82052e4d0, 0x7f87da01a4c8, 0x0, 0x7f87da01a4c8, 0xc820a50300, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/sock_posix.go:137 +0x351
net.socket(0x104c950, 0x3, 0x2, 0x1, 0x0, 0xc820a50300, 0x7f87da01a4c8, 0x0, 0x7f87da01a4c8, 0xc820a50300, ...)
	/usr/local/go/src/net/sock_posix.go:89 +0x411
net.internetSocket(0x104c950, 0x3, 0x7f87da01a4c8, 0x0, 0x7f87da01a4c8, 0xc820a50300, 0x0, 0xc800000000, 0x0, 0x1, ...)
	/usr/local/go/src/net/ipsock_posix.go:160 +0x141
net.dialTCP(0x104c950, 0x3, 0x0, 0xc820a50300, 0x0, 0xc800000000, 0x0, 0x1, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:171 +0x11e
net.dialSingle(0xc820c4a080, 0x7f87da01a438, 0xc820a50300, 0x0, 0x7fff00000000, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/dial.go:364 +0x3f5
net.dialSerial.func1(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/dial.go:336 +0x75
net.dial(0x104c950, 0x3, 0x7f87da01a438, 0xc820a50300, 0xc8203876a0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/usr/local/go/src/net/fd_unix.go:40 +0x60
net.dialSerial(0xc820c4a080, 0xc820a1a380, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/dial.go:338 +0x760
net.(*Dialer).Dial(0xc820387918, 0x104c950, 0x3, 0xc8201b8000, 0x16, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/dial.go:232 +0x50f
net.Dial(0x104c950, 0x3, 0xc8201b8000, 0x16, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/dial.go:186 +0x95
net/http.(*Transport).dial(0x1751640, 0x104c950, 0x3, 0xc8201b8000, 0x16, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:501 +0xc4
net/http.(*Transport).dialConn(0x1751640, 0x0, 0xc82093c450, 0x4, 0xc8201b8000, 0x16, 0xc820174f91, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:596 +0x19a9
net/http.(*Transport).getConn.func4(0x1751640, 0x0, 0xc82093c450, 0x4, 0xc8201b8000, 0x16, 0xc82043a060)
	/usr/local/go/src/net/http/transport.go:549 +0x66
created by net/http.(*Transport).getConn
	/usr/local/go/src/net/http/transport.go:551 +0x265

goroutine 33675 [syscall]:
syscall.Syscall(0x0, 0x7, 0xc820176800, 0x200, 0x0, 0x487, 0x412749)
	/usr/local/go/src/syscall/asm_linux_amd64.s:18 +0x5
syscall.read(0x7, 0xc820176800, 0x200, 0x200, 0x0, 0x0, 0x0)
	/usr/local/go/src/syscall/zsyscall_linux_amd64.go:783 +0x5f
syscall.Read(0x7, 0xc820176800, 0x200, 0x200, 0xc82098e4e8, 0x0, 0x0)
	/usr/local/go/src/syscall/syscall_unix.go:160 +0x4d
os.(*File).read(0xc82003a0f0, 0xc820176800, 0x200, 0x200, 0xc820176800, 0x0, 0x0)
	/usr/local/go/src/os/file_unix.go:211 +0x53
os.(*File).Read(0xc82003a0f0, 0xc820176800, 0x200, 0x200, 0x1, 0x0, 0x0)
	/usr/local/go/src/os/file.go:95 +0x8a
bytes.(*Buffer).ReadFrom(0xc8200d0070, 0x7f87da013298, 0xc82003a0f0, 0x0, 0x0, 0x0)
	/usr/local/go/src/bytes/buffer.go:173 +0x23f
io.copyBuffer(0x7f87da014120, 0xc8200d0070, 0x7f87da013298, 0xc82003a0f0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/io/io.go:375 +0x180
io.Copy(0x7f87da014120, 0xc8200d0070, 0x7f87da013298, 0xc82003a0f0, 0x0, 0x0, 0x0)
	/usr/local/go/src/io/io.go:351 +0x64
os/exec.(*Cmd).writerDescriptor.func1(0x0, 0x0)
	/usr/local/go/src/os/exec/exec.go:232 +0x8b
os/exec.(*Cmd).Start.func1(0xc820aee140, 0xc820567260)
	/usr/local/go/src/os/exec/exec.go:340 +0x1d
created by os/exec.(*Cmd).Start
	/usr/local/go/src/os/exec/exec.go:341 +0x96d

goroutine 33647 [syscall]:
syscall.Syscall6(0x3d, 0x296d, 0xc820b177ac, 0x0, 0xc8204345a0, 0x0, 0x0, 0xc820b17780, 0xc820b17788, 0xc820b17778)
	/usr/local/go/src/syscall/asm_linux_amd64.s:44 +0x5
syscall.wait4(0x296d, 0xc820b177ac, 0x0, 0xc8204345a0, 0x90, 0x0, 0x0)
	/usr/local/go/src/syscall/zsyscall_linux_amd64.go:172 +0x72
syscall.Wait4(0x296d, 0xc820b177f4, 0x0, 0xc8204345a0, 0xc82003a190, 0x0, 0x0)
	/usr/local/go/src/syscall/syscall_linux.go:256 +0x55
os.(*Process).wait(0xc820418240, 0x15, 0x0, 0x0)
	/usr/local/go/src/os/exec_unix.go:22 +0x105
os.(*Process).Wait(0xc820418240, 0x0, 0x0, 0x0)
	/usr/local/go/src/os/doc.go:45 +0x2d
os/exec.(*Cmd).Wait(0xc820aee140, 0x0, 0x0)
	/usr/local/go/src/os/exec/exec.go:380 +0x211
os/exec.(*Cmd).Run(0xc820aee140, 0x0, 0x0)
	/usr/local/go/src/os/exec/exec.go:258 +0x64
os/exec.(*Cmd).Output(0xc820aee140, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/os/exec/exec.go:408 +0x1a1
github.com/shirou/gopsutil/host.getLSB(0x0, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/shirou/gopsutil/host/host_linux.go:166 +0x56a
github.com/shirou/gopsutil/host.GetPlatformInformation(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/shirou/gopsutil/host/host_linux.go:194 +0x86
github.com/shirou/gopsutil/host.HostInfo(0xc8205671c0, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/shirou/gopsutil/host/host_linux.go:42 +0xd2
github.com/influxdata/telegraf/plugins/inputs/system.(*SystemStats).Gather(0x1795760, 0x7f87d393ba40, 0xc82018aa80, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/system/system.go:29 +0x76
github.com/influxdata/telegraf/agent.(*Agent).gatherParallel.func1(0xc820aea040, 0xc820154480, 0xc8201c4038, 0x0, 0xc820015410)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:137 +0x432
created by github.com/influxdata/telegraf/agent.(*Agent).gatherParallel
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/agent/agent.go:141 +0x442
```

config:

```
[[inputs.apache]]
        urls = ["http://shiseido1.odiso.net/server-status?auto"]

[[inputs.phpfpm]]
    urls = [
        "/var/run/default.sock:fpm-status",
        "/var/run/site1530.sock:fpm-status",
        "/var/run/site1559.sock:fpm-status",
        "/var/run/site1719.sock:fpm-status",
        "/var/run/site1741.sock:fpm-status",
        "/var/run/site1793.sock:fpm-status",
        "/var/run/site1826.sock:fpm-status",
        "/var/run/site1827.sock:fpm-status",
        "/var/run/site1828.sock:fpm-status",
        "/var/run/site1829.sock:fpm-status",
        "/var/run/site1830.sock:fpm-status",
        "/var/run/site1831.sock:fpm-status",
        "/var/run/site1832.sock:fpm-status",
        "/var/run/site1833.sock:fpm-status",
        "/var/run/site1424.sock:fpm-status",
        "/var/run/site2048.sock:fpm-status",
        "/var/run/site2049.sock:fpm-status",
        "/var/run/site2122.sock:fpm-status",
        "/var/run/site2298.sock:fpm-status",
        "/var/run/site2306.sock:fpm-status",
        "/var/run/site2307.sock:fpm-status",
        "/var/run/site2308.sock:fpm-status",
    ]

[[inputs.memcached]]
        servers = ["localhost:11211"]
[[inputs.cpu]]
        percpu = true
        totalcpu = true
        drop = ["time_*"]
[[inputs.disk]]
#       mount_points=["/"]
[[inputs.diskio]]
#       devices = ["sda", "sdb"]
[[inputs.mem]]
[[inputs.mysql]]
    servers = ["login:password@tcp(127.0.0.1:3306)/"]
[[inputs.net]]
#       interfaces = ["eth0", ... ]
[[inputs.netstat]]
[[inputs.swap]]
[[inputs.system]]


```

--- a/plugins/inputs/phpfpm/child.go
+++ b/plugins/inputs/phpfpm/child.go
@@ -1,0 +1,331 @@
+// Copyright 2011 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package phpfpm
+
+// This file implements FastCGI from the perspective of a child process.
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/http/cgi"
+	"os"
+	"strings"
+	"sync"
+	"time"
+)
+
+// request holds the state for an in-progress request. As soon as it's complete,
+// it's converted to an http.Request.
+type request struct {
+	pw        *io.PipeWriter
+	reqId     uint16
+	params    map[string]string
+	buf       [1024]byte
+	rawParams []byte
+	keepConn  bool
+}
+
+func newRequest(reqId uint16, flags uint8) *request {
+	r := &request{
+		reqId:    reqId,
+		params:   map[string]string{},
+		keepConn: flags&flagKeepConn != 0,
+	}
+	r.rawParams = r.buf[:0]
+	return r
+}
+
+// parseParams reads an encoded []byte into Params.
+func (r *request) parseParams() {
+	text := r.rawParams
+	r.rawParams = nil
+	for len(text) > 0 {
+		keyLen, n := readSize(text)
+		if n == 0 {
+			return
+		}
+		text = text[n:]
+		valLen, n := readSize(text)
+		if n == 0 {
+			return
+		}
+		text = text[n:]
+		if int(keyLen)+int(valLen) > len(text) {
+			return
+		}
+		key := readString(text, keyLen)
+		text = text[keyLen:]
+		val := readString(text, valLen)
+		text = text[valLen:]
+		r.params[key] = val
+	}
+}
+
+// response implements http.ResponseWriter.
+type response struct {
+	req         *request
+	header      http.Header
+	w           *bufWriter
+	wroteHeader bool
+}
+
+func newResponse(c *child, req *request) *response {
+	return &response{
+		req:    req,
+		header: http.Header{},
+		w:      newWriter(c.conn, typeStdout, req.reqId),
+	}
+}
+
+func (r *response) Header() http.Header {
+	return r.header
+}
+
+func (r *response) Write(data []byte) (int, error) {
+	if !r.wroteHeader {
+		r.WriteHeader(http.StatusOK)
+	}
+	return r.w.Write(data)
+}
+
+func (r *response) WriteHeader(code int) {
+	if r.wroteHeader {
+		return
+	}
+	r.wroteHeader = true
+	if code == http.StatusNotModified {
+		// Must not have body.
+		r.header.Del("Content-Type")
+		r.header.Del("Content-Length")
+		r.header.Del("Transfer-Encoding")
+	} else if r.header.Get("Content-Type") == "" {
+		r.header.Set("Content-Type", "text/html; charset=utf-8")
+	}
+
+	if r.header.Get("Date") == "" {
+		r.header.Set("Date", time.Now().UTC().Format(http.TimeFormat))
+	}
+
+	fmt.Fprintf(r.w, "Status: %d %s\r\n", code, http.StatusText(code))
+	r.header.Write(r.w)
+	r.w.WriteString("\r\n")
+}
+
+func (r *response) Flush() {
+	if !r.wroteHeader {
+		r.WriteHeader(http.StatusOK)
+	}
+	r.w.Flush()
+}
+
+func (r *response) Close() error {
+	r.Flush()
+	return r.w.Close()
+}
+
+type child struct {
+	conn    *conn
+	handler http.Handler
+
+	mu       sync.Mutex          // protects requests:
+	requests map[uint16]*request // keyed by request ID
+}
+
+func newChild(rwc io.ReadWriteCloser, handler http.Handler) *child {
+	return &child{
+		conn:     newConn(rwc),
+		handler:  handler,
+		requests: make(map[uint16]*request),
+	}
+}
+
+func (c *child) serve() {
+	defer c.conn.Close()
+	defer c.cleanUp()
+	var rec record
+	for {
+		if err := rec.read(c.conn.rwc); err != nil {
+			return
+		}
+		if err := c.handleRecord(&rec); err != nil {
+			return
+		}
+	}
+}
+
+var errCloseConn = errors.New("fcgi: connection should be closed")
+
+var emptyBody = ioutil.NopCloser(strings.NewReader(""))
+
+// ErrRequestAborted is returned by Read when a handler attempts to read the
+// body of a request that has been aborted by the web server.
+var ErrRequestAborted = errors.New("fcgi: request aborted by web server")
+
+// ErrConnClosed is returned by Read when a handler attempts to read the body of
+// a request after the connection to the web server has been closed.
+var ErrConnClosed = errors.New("fcgi: connection to web server closed")
+
+func (c *child) handleRecord(rec *record) error {
+	c.mu.Lock()
+	req, ok := c.requests[rec.h.Id]
+	c.mu.Unlock()
+	if !ok && rec.h.Type != typeBeginRequest && rec.h.Type != typeGetValues {
+		// The spec says to ignore unknown request IDs.
+		return nil
+	}
+
+	switch rec.h.Type {
+	case typeBeginRequest:
+		if req != nil {
+			// The server is trying to begin a request with the same ID
+			// as an in-progress request. This is an error.
+			return errors.New("fcgi: received ID that is already in-flight")
+		}
+
+		var br beginRequest
+		if err := br.read(rec.content()); err != nil {
+			return err
+		}
+		if br.role != roleResponder {
+			c.conn.writeEndRequest(rec.h.Id, 0, statusUnknownRole)
+			return nil
+		}
+		req = newRequest(rec.h.Id, br.flags)
+		c.mu.Lock()
+		c.requests[rec.h.Id] = req
+		c.mu.Unlock()
+		return nil
+	case typeParams:
+		// NOTE(eds): Technically a key-value pair can straddle the boundary
+		// between two packets. We buffer until we've received all parameters.
+		if len(rec.content()) > 0 {
+			req.rawParams = append(req.rawParams, rec.content()...)
+			return nil
+		}
+		req.parseParams()
+		return nil
+	case typeStdin:
+		content := rec.content()
+		if req.pw == nil {
+			var body io.ReadCloser
+			if len(content) > 0 {
+				// body could be an io.LimitReader, but it shouldn't matter
+				// as long as both sides are behaving.
+				body, req.pw = io.Pipe()
+			} else {
+				body = emptyBody
+			}
+			go c.serveRequest(req, body)
+		}
+		if len(content) > 0 {
+			// TODO(eds): This blocks until the handler reads from the pipe.
+			// If the handler takes a long time, it might be a problem.
+			req.pw.Write(content)
+		} else if req.pw != nil {
+			req.pw.Close()
+		}
+		return nil
+	case typeGetValues:
+		values := map[string]string{"FCGI_MPXS_CONNS": "1"}
+		c.conn.writePairs(typeGetValuesResult, 0, values)
+		return nil
+	case typeData:
+		// If the filter role is implemented, read the data stream here.
+		return nil
+	case typeAbortRequest:
+		c.mu.Lock()
+		delete(c.requests, rec.h.Id)
+		c.mu.Unlock()
+		c.conn.writeEndRequest(rec.h.Id, 0, statusRequestComplete)
+		if req.pw != nil {
+			req.pw.CloseWithError(ErrRequestAborted)
+		}
+		if !req.keepConn {
+			// connection will close upon return
+			return errCloseConn
+		}
+		return nil
+	default:
+		b := make([]byte, 8)
+		b[0] = byte(rec.h.Type)
+		c.conn.writeRecord(typeUnknownType, 0, b)
+		return nil
+	}
+}
+
+func (c *child) serveRequest(req *request, body io.ReadCloser) {
+	r := newResponse(c, req)
+	httpReq, err := cgi.RequestFromMap(req.params)
+	if err != nil {
+		// there was an error reading the request
+		r.WriteHeader(http.StatusInternalServerError)
+		c.conn.writeRecord(typeStderr, req.reqId, []byte(err.Error()))
+	} else {
+		httpReq.Body = body
+		c.handler.ServeHTTP(r, httpReq)
+	}
+	r.Close()
+	c.mu.Lock()
+	delete(c.requests, req.reqId)
+	c.mu.Unlock()
+	c.conn.writeEndRequest(req.reqId, 0, statusRequestComplete)
+
+	// Consume the entire body, so the host isn't still writing to
+	// us when we close the socket below in the !keepConn case,
+	// otherwise we'd send a RST. (golang.org/issue/4183)
+	// TODO(bradfitz): also bound this copy in time. Or send
+	// some sort of abort request to the host, so the host
+	// can properly cut off the client sending all the data.
+	// For now just bound it a little and
+	io.CopyN(ioutil.Discard, body, 100<<20)
+	body.Close()
+
+	if !req.keepConn {
+		c.conn.Close()
+	}
+}
+
+func (c *child) cleanUp() {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	for _, req := range c.requests {
+		if req.pw != nil {
+			// race with call to Close in c.serveRequest doesn't matter because
+			// Pipe(Reader|Writer).Close are idempotent
+			req.pw.CloseWithError(ErrConnClosed)
+		}
+	}
+}
+
+// Serve accepts incoming FastCGI connections on the listener l, creating a new
+// goroutine for each. The goroutine reads requests and then calls handler
+// to reply to them.
+// If l is nil, Serve accepts connections from os.Stdin.
+// If handler is nil, http.DefaultServeMux is used.
+func Serve(l net.Listener, handler http.Handler) error {
+	if l == nil {
+		var err error
+		l, err = net.FileListener(os.Stdin)
+		if err != nil {
+			return err
+		}
+		defer l.Close()
+	}
+	if handler == nil {
+		handler = http.DefaultServeMux
+	}
+	for {
+		rw, err := l.Accept()
+		if err != nil {
+			return err
+		}
+		c := newChild(rw, handler)
+		go c.serve()
+	}
+}

--- a/plugins/inputs/phpfpm/fcgi.go
+++ b/plugins/inputs/phpfpm/fcgi.go
@@ -17,11 +17,6 @@ import (
 	"errors"
 	"io"
 	"sync"
-
-	"net"
-	"strconv"
-
-	"strings"
 )
 
 // recType is a record type, as defined by
@@ -276,75 +271,4 @@ func (w *streamWriter) Write(p []byte) (int, error) {
 func (w *streamWriter) Close() error {
 	// send empty record to close the stream
 	return w.c.writeRecord(w.recType, w.reqId, nil)
-}
-
-func NewClient(h string, args ...interface{}) (fcgi *conn, err error) {
-	var con net.Conn
-	if len(args) != 1 {
-		err = errors.New("fcgi: not enough params")
-		return
-	}
-	switch args[0].(type) {
-	case int:
-		addr := h + ":" + strconv.FormatInt(int64(args[0].(int)), 10)
-		con, err = net.Dial("tcp", addr)
-	case string:
-		laddr := net.UnixAddr{Name: args[0].(string), Net: h}
-		con, err = net.DialUnix(h, nil, &laddr)
-	default:
-		err = errors.New("fcgi: we only accept int (port) or string (socket) params.")
-	}
-	fcgi = &conn{
-		rwc: con,
-	}
-	return
-}
-
-func (client *conn) Request(env map[string]string, requestData string) (retout []byte, reterr []byte, err error) {
-	defer client.rwc.Close()
-	var reqId uint16 = 1
-
-	err = client.writeBeginRequest(reqId, uint16(roleResponder), 0)
-	if err != nil {
-		return
-	}
-
-	err = client.writePairs(typeParams, reqId, env)
-	if err != nil {
-		return
-	}
-
-	if len(requestData) > 0 {
-		if err = client.writeRecord(typeStdin, reqId, []byte(requestData)); err != nil {
-			return
-		}
-	}
-
-	rec := &record{}
-	var err1 error
-
-	// recive untill EOF or FCGI_END_REQUEST
-READ_LOOP:
-	for {
-		err1 = rec.read(client.rwc)
-		if err1 != nil && strings.Contains(err1.Error(), "use of closed network connection") {
-			if err1 != io.EOF {
-				err = err1
-			}
-			break
-		}
-
-		switch {
-		case rec.h.Type == typeStdout:
-			retout = append(retout, rec.content()...)
-		case rec.h.Type == typeStderr:
-			reterr = append(reterr, rec.content()...)
-		case rec.h.Type == typeEndRequest:
-			fallthrough
-		default:
-			break READ_LOOP
-		}
-	}
-
-	return
 }

--- a/plugins/inputs/phpfpm/fcgi_client.go
+++ b/plugins/inputs/phpfpm/fcgi_client.go
@@ -1,0 +1,86 @@
+package phpfpm
+
+import (
+	"errors"
+	"io"
+	"net"
+	"strconv"
+	"strings"
+)
+
+// Create an fcgi client
+func newFcgiClient(h string, args ...interface{}) (*conn, error) {
+	var con net.Conn
+	if len(args) != 1 {
+		return nil, errors.New("fcgi: not enough params")
+	}
+
+	var err error
+	switch args[0].(type) {
+	case int:
+		addr := h + ":" + strconv.FormatInt(int64(args[0].(int)), 10)
+		con, err = net.Dial("tcp", addr)
+	case string:
+		laddr := net.UnixAddr{Name: args[0].(string), Net: h}
+		con, err = net.DialUnix(h, nil, &laddr)
+	default:
+		err = errors.New("fcgi: we only accept int (port) or string (socket) params.")
+	}
+	fcgi := &conn{
+		rwc: con,
+	}
+
+	return fcgi, err
+}
+
+func (client *conn) Request(
+	env map[string]string,
+	requestData string,
+) (retout []byte, reterr []byte, err error) {
+	defer client.rwc.Close()
+	var reqId uint16 = 1
+
+	err = client.writeBeginRequest(reqId, uint16(roleResponder), 0)
+	if err != nil {
+		return
+	}
+
+	err = client.writePairs(typeParams, reqId, env)
+	if err != nil {
+		return
+	}
+
+	if len(requestData) > 0 {
+		if err = client.writeRecord(typeStdin, reqId, []byte(requestData)); err != nil {
+			return
+		}
+	}
+
+	rec := &record{}
+	var err1 error
+
+	// recive untill EOF or FCGI_END_REQUEST
+READ_LOOP:
+	for {
+		err1 = rec.read(client.rwc)
+		if err1 != nil && strings.Contains(err1.Error(), "use of closed network connection") {
+			if err1 != io.EOF {
+				err = err1
+			}
+			break
+		}
+
+		switch {
+		case rec.h.Type == typeStdout:
+			retout = append(retout, rec.content()...)
+		case rec.h.Type == typeStderr:
+			reterr = append(reterr, rec.content()...)
+		case rec.h.Type == typeEndRequest:
+			fallthrough
+		default:
+			break READ_LOOP
+		}
+	}
+
+	return
+}

--- a/plugins/inputs/phpfpm/fcgi_test.go
+++ b/plugins/inputs/phpfpm/fcgi_test.go
@@ -1,0 +1,280 @@
+// Copyright 2011 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package phpfpm
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"testing"
+)
+
+var sizeTests = []struct {
+	size  uint32
+	bytes []byte
+}{
+	{0, []byte{0x00}},
+	{127, []byte{0x7F}},
+	{128, []byte{0x80, 0x00, 0x00, 0x80}},
+	{1000, []byte{0x80, 0x00, 0x03, 0xE8}},
+	{33554431, []byte{0x81, 0xFF, 0xFF, 0xFF}},
+}
+
+func TestSize(t *testing.T) {
+	b := make([]byte, 4)
+	for i, test := range sizeTests {
+		n := encodeSize(b, test.size)
+		if !bytes.Equal(b[:n], test.bytes) {
+			t.Errorf("%d expected %x, encoded %x", i, test.bytes, b)
+		}
+		size, n := readSize(test.bytes)
+		if size != test.size {
+			t.Errorf("%d expected %d, read %d", i, test.size, size)
+		}
+		if len(test.bytes) != n {
+			t.Errorf("%d did not consume all the bytes", i)
+		}
+	}
+}
+
+var streamTests = []struct {
+	desc    string
+	recType recType
+	reqId   uint16
+	content []byte
+	raw     []byte
+}{
+	{"single record", typeStdout, 1, nil,
+		[]byte{1, byte(typeStdout), 0, 1, 0, 0, 0, 0},
+	},
+	// this data will have to be split into two records
+	{"two records", typeStdin, 300, make([]byte, 66000),
+		bytes.Join([][]byte{
+			// header for the first record
+			{1, byte(typeStdin), 0x01, 0x2C, 0xFF, 0xFF, 1, 0},
+			make([]byte, 65536),
+			// header for the second
+			{1, byte(typeStdin), 0x01, 0x2C, 0x01, 0xD1, 7, 0},
+			make([]byte, 472),
+			// header for the empty record
+			{1, byte(typeStdin), 0x01, 0x2C, 0, 0, 0, 0},
+		},
+			nil),
+	},
+}
+
+type nilCloser struct {
+	io.ReadWriter
+}
+
+func (c *nilCloser) Close() error { return nil }
+
+func TestStreams(t *testing.T) {
+	var rec record
+outer:
+	for _, test := range streamTests {
+		buf := bytes.NewBuffer(test.raw)
+		var content []byte
+		for buf.Len() > 0 {
+			if err := rec.read(buf); err != nil {
+				t.Errorf("%s: error reading record: %v", test.desc, err)
+				continue outer
+			}
+			content = append(content, rec.content()...)
+		}
+		if rec.h.Type != test.recType {
+			t.Errorf("%s: got type %d expected %d", test.desc, rec.h.Type, test.recType)
+			continue
+		}
+		if rec.h.Id != test.reqId {
+			t.Errorf("%s: got request ID %d expected %d", test.desc, rec.h.Id, test.reqId)
+			continue
+		}
+		if !bytes.Equal(content, test.content) {
+			t.Errorf("%s: read wrong content", test.desc)
+			continue
+		}
+		buf.Reset()
+		c := newConn(&nilCloser{buf})
+		w := newWriter(c, test.recType, test.reqId)
+		if _, err := w.Write(test.content); err != nil {
+			t.Errorf("%s: error writing record: %v", test.desc, err)
+			continue
+		}
+		if err := w.Close(); err != nil {
+			t.Errorf("%s: error closing stream: %v", test.desc, err)
+			continue
+		}
+		if !bytes.Equal(buf.Bytes(), test.raw) {
+			t.Errorf("%s: wrote wrong content", test.desc)
+		}
+	}
+}
+
+type writeOnlyConn struct {
+	buf []byte
+}
+
+func (c *writeOnlyConn) Write(p []byte) (int, error) {
+	c.buf = append(c.buf, p...)
+	return len(p), nil
+}
+
+func (c *writeOnlyConn) Read(p []byte) (int, error) {
+	return 0, errors.New("conn is write-only")
+}
+
+func (c *writeOnlyConn) Close() error {
+	return nil
+}
+
+func TestGetValues(t *testing.T) {
+	var rec record
+	rec.h.Type = typeGetValues
+
+	wc := new(writeOnlyConn)
+	c := newChild(wc, nil)
+	err := c.handleRecord(&rec)
+	if err != nil {
+		t.Fatalf("handleRecord: %v", err)
+	}
+
+	const want = "\x01\n\x00\x00\x00\x12\x06\x00" +
+		"\x0f\x01FCGI_MPXS_CONNS1" +
+		"\x00\x00\x00\x00\x00\x00\x01\n\x00\x00\x00\x00\x00\x00"
+	if got := string(wc.buf); got != want {
+		t.Errorf(" got: %q\nwant: %q\n", got, want)
+	}
+}
+
+func nameValuePair11(nameData, valueData string) []byte {
+	return bytes.Join(
+		[][]byte{
+			{byte(len(nameData)), byte(len(valueData))},
+			[]byte(nameData),
+			[]byte(valueData),
+		},
+		nil,
+	)
+}
+
+func makeRecord(
+	recordType recType,
+	requestId uint16,
+	contentData []byte,
+) []byte {
+	requestIdB1 := byte(requestId >> 8)
+	requestIdB0 := byte(requestId)
+
+	contentLength := len(contentData)
+	contentLengthB1 := byte(contentLength >> 8)
+	contentLengthB0 := byte(contentLength)
+	return bytes.Join([][]byte{
+		{1, byte(recordType), requestIdB1, requestIdB0, contentLengthB1,
+			contentLengthB0, 0, 0},
+		contentData,
+	},
+		nil)
+}
+
+// a series of FastCGI records that start a request and begin sending the
+// request body
+var streamBeginTypeStdin = bytes.Join([][]byte{
+	// set up request 1
+	makeRecord(typeBeginRequest, 1,
+		[]byte{0, byte(roleResponder), 0, 0, 0, 0, 0, 0}),
+	// add required parameters to request 1
+	makeRecord(typeParams, 1, nameValuePair11("REQUEST_METHOD", "GET")),
+	makeRecord(typeParams, 1, nameValuePair11("SERVER_PROTOCOL", "HTTP/1.1")),
+	makeRecord(typeParams, 1, nil),
+	// begin sending body of request 1
+	makeRecord(typeStdin, 1, []byte("0123456789abcdef")),
+},
+	nil)
+
+var cleanUpTests = []struct {
+	input []byte
+	err   error
+}{
+	// confirm that child.handleRecord closes req.pw after aborting req
+	{
+		bytes.Join([][]byte{
+			streamBeginTypeStdin,
+			makeRecord(typeAbortRequest, 1, nil),
+		},
+			nil),
+		ErrRequestAborted,
+	},
+	// confirm that child.serve closes all pipes after error reading record
+	{
+		bytes.Join([][]byte{
+			streamBeginTypeStdin,
+			nil,
+		},
+			nil),
+		ErrConnClosed,
+	},
+}
+
+type nopWriteCloser struct {
+	io.ReadWriter
+}
+
+func (nopWriteCloser) Close() error {
+	return nil
+}
+
+// Test that child.serve closes the bodies of aborted requests and closes the
+// bodies of all requests before returning. Causes deadlock if either condition
+// isn't met. See issue 6934.
+func TestChildServeCleansUp(t *testing.T) {
+	for _, tt := range cleanUpTests {
+		input := make([]byte, len(tt.input))
+		copy(input, tt.input)
+		rc := nopWriteCloser{bytes.NewBuffer(input)}
+		done := make(chan bool)
+		c := newChild(rc, http.HandlerFunc(func(
+			w http.ResponseWriter,
+			r *http.Request,
+		) {
+			// block on reading body of request
+			_, err := io.Copy(ioutil.Discard, r.Body)
+			if err != tt.err {
+				t.Errorf("Expected %#v, got %#v", tt.err, err)
+			}
+			// not reached if body of request isn't closed
+			done <- true
+		}))
+		go c.serve()
+		// wait for body of request to be closed or all goroutines to block
+		<-done
+	}
+}
+
+type rwNopCloser struct {
+	io.Reader
+	io.Writer
+}
+
+func (rwNopCloser) Close() error {
+	return nil
+}
+
+// Verifies it doesn't crash. 	Issue 11824.
+func TestMalformedParams(t *testing.T) {
+	input := []byte{
+		// beginRequest, requestId=1, contentLength=8, role=1, keepConn=1
+		1, 1, 0, 1, 0, 8, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0,
+		// params, requestId=1, contentLength=10, k1Len=50, v1Len=50 (malformed, wrong length)
+		1, 4, 0, 1, 0, 10, 0, 0, 50, 50, 3, 4, 5, 6, 7, 8, 9, 10,
+		// end of params
+		1, 4, 0, 1, 0, 0, 0, 0,
+	}
+	rw := rwNopCloser{bytes.NewReader(input), ioutil.Discard}
+	c := newChild(rw, http.DefaultServeMux)
+	c.serve()
+}

--- a/plugins/inputs/phpfpm/phpfpm.go
+++ b/plugins/inputs/phpfpm/phpfpm.go
@@ -112,6 +112,7 @@ func (g *phpfpm) gatherServer(addr string, acc telegraf.Accumulator) error {
 		statusPath string
 	)
 
+	var err error
 	if strings.HasPrefix(addr, "fcgi://") || strings.HasPrefix(addr, "cgi://") {
 		u, err := url.Parse(addr)
 		if err != nil {
@@ -120,7 +121,7 @@ func (g *phpfpm) gatherServer(addr string, acc telegraf.Accumulator) error {
 		socketAddr := strings.Split(u.Host, ":")
 		fcgiIp := socketAddr[0]
 		fcgiPort, _ := strconv.Atoi(socketAddr[1])
-		fcgi, _ = NewClient(fcgiIp, fcgiPort)
+		fcgi, err = newFcgiClient(fcgiIp, fcgiPort)
 	} else {
 		socketAddr := strings.Split(addr, ":")
 		if len(socketAddr) >= 2 {
@@ -134,8 +135,13 @@ func (g *phpfpm) gatherServer(addr string, acc telegraf.Accumulator) error {
 		if _, err := os.Stat(socketPath); os.IsNotExist(err) {
 			return fmt.Errorf("Socket doesn't exist  '%s': %s", socketPath, err)
 		}
-		fcgi, _ = NewClient("unix", socketPath)
+		fcgi, err = newFcgiClient("unix", socketPath)
 	}
+
+	if err != nil {
+		return err
+	}
+
 	return g.gatherFcgi(fcgi, statusPath, acc)
 }
 

